sample/src/test/kotlin/dev/afanasev/kson/tests/GeneratedAdaptersTest.kt

```diff
@@ -228,6 +228,42 @@ class GeneratedAdaptersTest {
         assertThat(adapter).contains(REFLECTIVE_TYPE_ADAPTER_FACTORY_NAME)
     }
 
+    @Test
+    fun `should use alternate field from annotation`() {
+        val json = """
+            {
+                "customKey1": "first",
+                "key2": "second"
+            }
+            """
+
+        val entity = gson.fromJson(json, EntityWithCustomKeys::class.java)
+
+        assertThat(entity.key1).isEqualTo("first")
```

Comment:
If you are trying to check if you adapter can use alternate keys, why are you also trying to check if it can use primary key in exact same method?